scipy/stats/tests/test_morestats.py

```diff
@@ -2076,6 +2076,32 @@ def test_user_defined_optimizer_and_brack_raises_error(self):
             stats.boxcox_normmax(self.x, brack=(-2.0, 2.0),
                                  optimizer=optimizer)
 
+    def test_negative_ymax(self):
+        with pytest.raises(ValueError, match="`ymax` must be strictly positive"):
+            stats.boxcox_normmax(self.x, ymax=-1)
+
+    @pytest.mark.parametrize('x', [
+        # Attempt to trigger overflow in power expressions.
+        np.array([2003.0, 1950.0, 1997.0, 2000.0, 2009.0,
+                  2009.0, 1980.0, 1999.0, 2007.0, 1991.0]),
+        # Attempt to trigger overflow with a large optimal lambda.
+        np.array([2003.0, 1950.0, 1997.0, 2000.0, 2009.0]),
+        # Attempt to trigger overflow with large data.
+        np.array([2003.0e200, 1950.0e200, 1997.0e200, 2000.0e200, 2009.0e200])
+    ])
+    @pytest.mark.parametrize("ymax", [1e10, 1e30, 1e100])
+    # TODO: add method "pearsonr" after fix overflow issue
+    @pytest.mark.parametrize("method", ["mle"])
+    def test_user_defined_ymax(self, x, ymax, method):
+        # Use ymax=None and assert that there is indeed an overflow.
+        # Then continue with the rest of this test.
+        with pytest.warns(UserWarning, match="The optimal lambda is"):
+            stats.boxcox_normmax(x, ymax=None, method=method)
+
+        lmb = stats.boxcox_normmax(x, ymax=ymax, method=method)
+        ymax_res = np.max(stats.boxcox(x, lmb))
+        assert_allclose(ymax, ymax_res, rtol=1e-1)
```

Comment:
Based on my other comments, I think you can just include default `ymax` in the parametrize cases. You should always get the warning, and the maximum of the transformed data should be close to `ymax` (or whatever `ymax` turns into, in case of the default.

Please also test float32 since behavior is dtype dependent.

I think there should be a separate test for `ymax=np.inf`. There shouldn't be a warning, and you should check the returned optimal `lmbda` against ground truth. To keep things easy, maybe you can find a test case for which you'd get overflow with `float32` input but not `float64`, then you'd just check that the returned `lmbda` is the same for either `float32` or `float64` input. (Does this make sense?)

Reply:
I agree to add a test for the default `ymax`, dtype of float32, and `ymax=np.inf`.

> maybe you can find a test case for which you'd get overflow with float32 input but not float64

Since the ground truth is not easy to test, I think it's reasonable to test like this.